velox/functions/prestosql/ArrayIntersectExcept.cpp

```diff
@@ -269,29 +271,143 @@ class ArrayIntersectExceptFunction : public exec::VectorFunction {
 
   // If there's a `constantSet`, whether it refers to left or right-hand side.
   const bool isLeftConstant_{false};
-};
+}; // class ArrayIntersectExcept
+
+template <typename T>
+class ArraysOverlapFunction : public exec::VectorFunction {
+ public:
+  ArraysOverlapFunction() {}
+
+  explicit ArraysOverlapFunction(
+      SetWithNull<T> constantSet,
+      bool isLeftConstant)
+      : constantSet_(std::move(constantSet)), isLeftConstant_(isLeftConstant) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx* context,
+      VectorPtr* result) const override {
+    // if one of them is a ConstantVector.
+    // Iterate through the other ArrayVector->elementsFlatVector
+    // and do constantSet_.find()
+    BaseVector* left = args[0].get();
+    BaseVector* right = args[1].get();
+    if (constantSet_.has_value() && isLeftConstant_) {
+      std::swap(left, right);
+    }
+    exec::LocalDecodedVector decoder(context, *left, rows);
+    exec::LocalDecodedVector elementsDecoder(context);
+    auto decodedLeftElements =
+        getDecodedElementsFromArrayVector(decoder, elementsDecoder, rows);
+
+    BaseVector::ensureWritable(rows, BOOLEAN(), context->pool(), result);
+    auto resBoolVector = (*result)->template asFlatVector<bool>();
+    auto processRow = [&](const vector_size_t rowId,
+                          const SetWithNull<T>& rightSet) {
+      auto decodedLeftArray = decoder.get();
+      auto baseLeftArray = decodedLeftArray->base()->as<ArrayVector>();
+      auto idx = decodedLeftArray->index(rowId);
+      auto offset = baseLeftArray->offsetAt(idx);
+      auto size = baseLeftArray->sizeAt(idx);
+
+      for (auto i = offset; i < (offset + size); ++i) {
+        // For each element in the current row search for it in the rightSet.
+        if (decodedLeftElements->isNullAt(i)) {
+          // Arrays overlap skips null values.
+          continue;
+        }
+        if (rightSet.set.count(decodedLeftElements->valueAt<T>(i)) > 0) {
+          // Found and overlapping element. Add to result set.
+          resBoolVector->set(rowId, true);
+          return;
+        }
+      }
+      // If none of them is found in the rightSet, set false for current row
```

Comment:
> Returns null if there are no non-null elements in common but either array contains null.

I don't see where this logic is implemented. Can you give me a pointer? Is there a test for that?

Reply:
Yes, i missed that. Thanks for pointing out:

```
presto:default> select * from t1;
      c0      
--------------
 [1, 2, 3]    
 [5, 4, null] 
 [5, 4]       
(3 rows)

Query 20220216_000754_00015_pg2ci, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
318ms [3 rows, 315B] [9 rows/s, 990B/s]

presto:default> select arrays_overlap(c0, array[1,2]) from t1;
 _col0 
-------
 true  
 NULL  
 false 
(3 rows)
```

This how prestoDB returns for non-overlapping case with null and without null.

Reply:
I switched to using nulls to adhere to prestodb standard.